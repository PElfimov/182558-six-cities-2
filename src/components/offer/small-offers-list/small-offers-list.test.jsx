import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import SmallOffersList from './small-offers-list';
Enzyme.configure({adapter: new Adapter()});

jest.mock(`../../../mocks/test-mocks`);

const offers = [
  {
    isPremium: true,
    id: 1,
    city: {
      name: `barnaul`,
      coordinates: [53.550341, 10.000654],
      zoom: 13
    },
    cost: 120,
    name: `Beautiful & luxurious apartment at great location`,
    rating: 55,
    type: `Apartment`
  },
  {
    isPremium: false,
    id: 2,
    city: {
      name: `barnaul`,
      coordinates: [53.550341, 10.000654],
      zoom: 13
    },
    cost: 132,
    name: `Wood and stone place`,
    rating: 80,
    type: `Apartment`
  },
  {
    isPremium: true,
    id: 3,
    city: {
      name: `barnaul`,
      coordinates: [53.550341, 10.000654],
      zoom: 13
    },
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`
  },
  {
    isPremium: true,
    id: 4,
    city: {
      name: `barnaul`,
      coordinates: [53.550341, 10.000654],
      zoom: 13
    },
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`
  },
  {
    isPremium: true,
    id: 5,
    city: {
      name: `barnaul`,
      coordinates: [53.550341, 10.000654],
      zoom: 13
    },
    cost: 120,
    name: `Wood and stone place`,
    rating: 88,
    type: `Private room`
  }
];

it(`HotelCard correctly renders after relaunch`, () => {
  const tree = shallow(
      <SmallOffersList
        offers={offers}
        id={1} />
  );

  expect(toJSON(tree)).toMatchSnapshot();
});
