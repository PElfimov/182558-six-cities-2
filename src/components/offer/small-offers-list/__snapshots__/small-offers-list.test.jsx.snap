// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCard correctly renders after relaunch 1`] = `
<Fragment>
  <HotelCard
    key="2"
    offer={
      Object {
        "city": Object {
          "coordinates": Array [
            53.550341,
            10.000654,
          ],
          "name": "barnaul",
          "zoom": 13,
        },
        "cost": 132,
        "id": 2,
        "isPremium": false,
        "name": "Wood and stone place",
        "rating": 80,
        "type": "Apartment",
      }
    }
    onHover={[Function]}
  />
  <HotelCard
    key="3"
    offer={
      Object {
        "city": Object {
          "coordinates": Array [
            53.550341,
            10.000654,
          ],
          "name": "barnaul",
          "zoom": 13,
        },
        "cost": 120,
        "id": 3,
        "isPremium": true,
        "name": "Canal View Prinsengracht",
        "rating": 55,
        "type": "Apartment",
      }
    }
    onHover={[Function]}
  />
  <HotelCard
    key="4"
    offer={
      Object {
        "city": Object {
          "coordinates": Array [
            53.550341,
            10.000654,
          ],
          "name": "barnaul",
          "zoom": 13,
        },
        "cost": 80,
        "id": 4,
        "isPremium": true,
        "name": "Nice, cozy, warm big bed apartment",
        "rating": 100,
        "type": "Apartment",
      }
    }
    onHover={[Function]}
  />
</Fragment>
`;
